pkg/servicemesh/servicemesh-control-plane.go

```diff
@@ -0,0 +1,578 @@
+package servicemesh
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	istioV1 "maistra.io/api/core/v1"
+	istioV2 "maistra.io/api/core/v2"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ControlPlaneBuilder provides a struct for serviceMeshControlPlane object from the cluster and
+// a serviceMeshControlPlane definition.
+type ControlPlaneBuilder struct {
+	// serviceMeshControlPlane definition, used to create the serviceMeshControlPlane object.
+	Definition *istioV2.ServiceMeshControlPlane
+	// Created serviceMeshControlPlane object.
+	Object *istioV2.ServiceMeshControlPlane
+	// Used in functions that define or mutate serviceMeshControlPlane definition. errorMsg is processed
+	// before the serviceMeshControlPlane object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewServiceMeshControlPlane method creates new instance of builder.
+func NewServiceMeshControlPlane(apiClient *clients.Settings, name, nsname string) *ControlPlaneBuilder {
```

Comment:
Please re-name it to NewControlPlaneBuilder. We have serviceMesh in the name of the package no need to duplicate it